migtests/scripts/compare-html-reports.py

```diff
@@ -0,0 +1,161 @@
+#!/usr/bin/env python3
+
+import argparse
+import re
+from bs4 import BeautifulSoup
+import difflib
+import sys
+
+def extract_html_data(html_content):
+    """Extracts structured data from HTML content and normalizes it."""
+    soup = BeautifulSoup(html_content, 'html.parser')
+
+    def normalize_text(text):
+        # Replace non-breaking spaces and other similar characters
+        text = text.replace('\xa0', ' ')
+        text = text.replace('\u200B', '')
+        text = re.sub(r'\s+', ' ', text)
+        return text.strip()
+
+    def extract_texts(elements):
+        """Extract and normalize inner text while keeping structure."""
+        return [normalize_text(el.get_text(separator=" ")) for el in elements]
+
+    all_divs = soup.find_all("div")
+    filtered_divs = []
+    should_sort = False
+
+    for div in all_divs:
+        prev_h2 = div.find_previous("h2")
+        if prev_h2 and "Sharding Recommendations" in prev_h2.get_text():
+            should_sort = True
+
+        if "wrapper" not in div.get("class", []):  # Keep the wrapper exclusion
+            filtered_divs.append((div, should_sort))
+
+    # Extract and process div text
+    div_texts_sorted = []
+    for div, should_sort in filtered_divs:
+        div_text = normalize_text(div.get_text(separator=" ")).strip()
+        if should_sort:
+            div_text = " ".join(sorted(div_text.split()))  # Sort content within div
+        div_texts_sorted.append(div_text)
+
+    div_texts_sorted = sorted(div_texts_sorted) if any(s for _, s in filtered_divs) else div_texts_sorted
```

Comment:
please break the extract_html function further smaller functions for parsing divs, paragraphs etc..